117-add-scope-parameters-to-maven-artifact-for-kafka-connect-build.md

```diff
@@ -0,0 +1,101 @@
+# Add scope parameter to Maven artifacts for Kafka Connect build
+
+Kafka Connect build creates a container image using the plugins configured through the `.spec.build.plugins` property of the `KafkaConnect` custom resource.
+
+Plugin artifacts can be specified using Maven coordinates under `spec.build.plugins[].artifacts[]` to identify artifacts and dependencies so that they can be located and fetched during the build process.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      #...
+    plugins:
+      - name: my-plugin
+        artifacts:
+          - type: maven
+            repository: https://mvnrepository.com
+            group: <maven_group>
+            artifact: <maven_artifact>
+            version: <maven_version_number>
+```
+
+This proposal aims to support specifying the scope threshold for Maven artifacts, enabling users to configure which dependencies they would like to fetch and include in the container image.
+
+## Current situation
+
+Currently, no scope threshold is specified when pulling Maven dependencies, which results in downloading all dependencies, including test, provided, system, etc.
+
+This can make the build process longer, container images larger, and potentially lead to version conflicts as discussed in [issue #11799](https://github.com/strimzi/strimzi-kafka-operator/issues/11799).
+
+## Motivation
+
+We should allow users to specify the scope threshold for Maven artifacts to configure which dependencies they would like to include in the container image.
+
+Benefits include:
+- Faster build process
+- Smaller container images
+- Reduced risk of version conflicts
+- More control over the build process for users
+
+## Proposal
+
+The proposal is to allow users to specify a scope threshold for copying Maven dependencies during the Kafka Connect build, using a new property `scope` that applies only to Maven artifacts.
+
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaConnect
+metadata:
+  name: my-connect-cluster
+spec:
+  #...
+  build:
+    output:
+      #...
+    plugins:
+      - name: my-plugin
+        artifacts:
+          - type: maven
+            repository: https://mvnrepository.com
+            group: <maven_group>
+            artifact: <maven_artifact>
+            version: <maven_version_number>
+            scope: runtime # new optional property
```

Comment:
So, gicen the Maven options is `includeScope` and given there is an `excludeScope` which we do not add in this proposal but we might add it later ... should we name the API field `includeScope` as well?

Reply:
Thanks! I have updated the proposal to rename the parameter as suggested.